Mathlib/Geometry/Manifold/IntegralCurve.lean

```diff
@@ -199,14 +219,39 @@ lemma IsIntegralCurveOn.comp_add (hγ : IsIntegralCurveOn γ v s) (dt : ℝ) :
   simp only [mfld_simps, hasFDerivWithinAt_univ]
   exact HasFDerivAt.add_const (hasFDerivAt_id _) _
 
-lemma isIntegralCurveOn_comp_add {dt : ℝ} :
+lemma isIntegralCurveOn_comp_add (dt : ℝ) :
     IsIntegralCurveOn γ v s ↔ IsIntegralCurveOn (γ ∘ (· + dt)) v { t | t + dt ∈ s } := by
   refine ⟨fun hγ ↦ hγ.comp_add _, fun hγ ↦ ?_⟩
   convert hγ.comp_add (-dt)
   · ext t
     simp only [Function.comp_apply, neg_add_cancel_right]
   · simp only [mem_setOf_eq, neg_add_cancel_right, setOf_mem_eq]
 
+lemma isIntegralCurveOn_comp_sub (dt : ℝ) :
+    IsIntegralCurveOn γ v s ↔ IsIntegralCurveOn (γ ∘ (· - dt)) v { t | t - dt ∈ s } := by
+  refine ⟨fun hγ ↦ hγ.comp_add _, fun hγ ↦ ?_⟩
+  convert hγ.comp_add dt
+  ext t
+  · simp only [Function.comp_apply, add_sub_cancel]
+  · simp only [mem_setOf_eq, add_sub_cancel, setOf_mem_eq]
+
+-- TODO: missing lemma(s) for `Ioo`
```

Comment:
You should either add these lemmas or (if this is a red herring) excise this comment.

Note also that it may (or may not!) be useful to use the pointwise API. For example:
```lean
open Pointwise

example (a b dt : ℝ) : (-dt) +ᵥ Ioo a b = {t | t + dt ∈ Ioo a b} := by
  ext; simp [mem_neg_vadd_set_iff, add_comm dt]
```
etc.